src/sympc/session/session_manager.py

```diff
@@ -50,11 +43,19 @@ def setup_mpc(session: Session) -> None:
         Args:
             session (Session): Session to send.
         """
+        uuids: Dict[int, UUID] = {}
         for rank, party in enumerate(session.parties):
             # Assign a new rank before sending it to another party
-            session.rank = rank
-            session.session_ptrs.append(session.send(party))  # type: ignore
+            session_party = session.copy()
+            session_party.rank = rank
 
+            # And a new uuid
+            session_party.uuid = uuid4()
+            uuids[rank] = session_party.uuid
+            session.session_ptrs.append(session_party.send(party))  # type: ignore
+
```

Comment:
Q: The `set_session`  which maps parties `uuid` to Session objects, should it assigned it SessionManager class.